selector-set.js

```diff
@@ -321,15 +321,16 @@
               if (obj.selector === selector && (removeAll || obj.data === data)) {
                 objs.splice(k, 1);
                 removedIds[obj.id] = true;
+                removedSelectors[obj.selector] = true;
               }
             }
           }
           break;
         }
       }
     }
-
     this.size -= Object.keys(removedIds).length;
+    this.selectors = this.selectors.filter(function (selector) { return removedSelectors[selector] })
```

Comment:
I'd probably still like to avoid ES5 runtime features like `filter` in this project. I know it's super old, but I don't think any of this code makes use of them.